### ISSUE TITLE
Shutting down DocPad?

### ISSUE THREAD

**Issue by balupton** [docpad/docpad#1072]
Hi everyone.

I'm considering shutting down DocPad. Taking the website and servers down.

DocPad hasn't been maintained since 2014. Since the focuses on my life went in very different directions. And since funding switched from full-time to casual.

There are plenty of bug reports coming in from incompatibilities with newer versions of things.

It seems less cruel to take everything down so people aren't wasting their time on an unmaintained and buggy solution.

DocPad now can be replaced by custom promise scripts fairly easily - I would say most sites would be able to switch to a custom solution within a week.

Let's aim for February for this move.

**Comment by sulthan-betabulls** [comment]
Hi @balupton ,
We designed some websites using Docpad Framework. If you Shutdown the docpad, is it affect the sites using docpad?

Thanks in Advance...

**Comment by balupton** [comment]
Things that will go down:

- Skeleton selection when initialising a new project
- Error reporting
- Usage statistics
- Subscriber signups
- Version outdated checks
- Plugin compat checks

The DocPad server stuff is at:
https://github.com/bevry/helper/blob/master/source/plugins/docpad.js

**Comment by sulthan-betabulls** [comment]
That's Awesome. I am looking forward for Docpad v7...

**Comment by balupton** [comment]
> That's Awesome. I am looking forward for Docpad v7...

DocPad v7 isn't happening, hence the desire to shutdown the servers

**Comment by StephenBrown2** [comment]
I will be sad to see it go, but I understand the impetus. Has there been talk of handing off to another community leader, or a democracy of active contributors ( @SteveMcArthur perhaps)? Hugo has just gained image manipulation capabilities, so I am sure I can find a replacement either way.

**Comment by Aglezabad** [comment]
If I could, could I do a fork of Docpad? There's a lot of work and fixing bugs, but I want to do an attempt. But, what's your opinion? May be worth it or Docpad is connected very tight with the dependencies of other Bevry projects?

**Comment by mikeumus** [comment]
What's the server costs?

**Comment by balupton** [comment]
The servers cost $7/month, but currently are on free credits, so they can stay up.

The issue is more about putting an `unmaintained - no active maintainers` banner on the website to discourage new users.

**Comment by balupton** [comment]
> If I could, could I do a fork of Docpad? There's a lot of work and fixing bugs, but I want to do an attempt. But, what's your opinion? May be worth it or Docpad is connected very tight with the dependencies of other Bevry projects?

Well, a fork wouldn't even be necessary, happy to give you maintenance abilities on the existing repo.

The think is, 2016 was spent updating every single bevry project, they all got new releases. I updated DocPad to incorporate these updates on the https://github.com/docpad/docpad/tree/dev-updatedeps branch, however I found out that ecosystem updates (mostly with npm) broke the plugin testing, so I started working on that - updated about 60% of the plugins, but still more to go.

So the next 6.x release is around the corner - but there is no maintainers right now to have it happen, as they are on other projects - and there are alternatives to DocPad now that can also do the job, for instance even custom scripts are an option now.

The 7.x release will never happen, as it is about 6 months of work - it would establish DocPad as the best static site generator there is - but there is so many other more important issues that static site generation imho.

**Comment by balupton** [comment]
The alternative is people just wait for me to occasionally be motived to devote a day here and there to upgrading it. Perhaps this would even be at the end of the year.

**Comment by mikeumus** [comment]
I say we just archive it for now and focus on new projects. The project will still be here.

**Comment by balupton** [comment]
It is actually a bit motivating to do say one plugin each morning, just to get into the routine of a productive day. @Aglezabad if you do want to collab, or have me hand over the reigns, reach out to me on our slack channel, or hit me up on wire.com - username balupton

**Comment by Aglezabad** [comment]
Hi:
I don't have so much time (too many projects and I'm ending my degree), but I can get some hours to help you as much as I can.

Best regards.

**Comment by balupton** [comment]
Okay. The path to upgrading DocPad's ecosystem is as follows:

```
# install https://github.com/bevry/based
npm install -g bevry-base

git clone git@github.com:docpad/extras.git docpad-extras
cd docpad-extras
npm install
npm run clone
```

then inside each plugin within `./docpad-extras/plugins` run `bevry-base` to upgrade the project to the latest conventions (including CoffeeScript v2). [associatedfiles](https://github.com/docpad/docpad-plugin-associatedfiles) has already bene completed.

Once all plugins are upgraded, then we can release the new DocPad version.

**Comment by balupton** [comment]
Alright. v1.11.0 of bevry-base released that makes the process more streamlined.

Updated plugins:

- https://github.com/docpad/docpad-plugin-associatedfiles
- https://github.com/docpad/docpad-plugin-autoprefix

**Comment by Aglezabad** [comment]
Ok, I will contact you as soon as I can, so I could help you.

**Comment by balupton** [comment]
`bevry-base` can now scaffold new projects

**Comment by Aglezabad** [comment]
I've started with my docpad plugin, so I can get some practise in this process, but npm shows me that bevry-base is deprecated. 

```npm WARN deprecated bevry-base@1.13.0: `bevry-base` on NPM has moved to `boundation`, use that package instead for updates```

Should may I use boundation instead?

Regards.

**Comment by mikeumus** [comment]
@Aglezabad, yes it's just a renaming: https://github.com/bevry/boundation

**Comment by balupton** [comment]
Closing this issue, as will be maintained, just as a side project.

**Comment by balupton** [comment]
https://github.com/docpad/docpad/tree/dev-updatedeps just got a significant push

all that is left for the next minor to complete

- is updating the rest of the plugins with boundation, to make them coffeescript v2 based, however this is not necessary if the environment variable `EDITIONS_SYNTAX_BLACKLIST=esnext` is set

- merging in the `master` changes into `dev-updatedeps`

**Comment by balupton** [comment]
Okay, the next minor is released as a beta.

Pushed to the `beta` git branch, and released to `6.80.0-beta`, under the `beta` npm tag.
